skbio/diversity/_driver.py

```diff
@@ -180,9 +182,114 @@ def alpha_diversity(metric, counts, ids=None, validate=True, **kwargs):
     return pd.Series(results, index=ids)
 
 
+@deprecated(as_of='0.4.2-dev', until='0.5.0',
+            reason=('The return type is unstable. Developer caution is '
+                    'advised. The resulting DistanceMatrix object may include '
+                    'zeros when distance has not been calculated, and '
+                    'therefore can be misleading.'))
+def partial_beta_diversity(metric, counts, ids=None, validate=None,
```

Comment:
Let's step back. Is it desirable for `beta_diversity(*args, **kwargs) == partial_beta_diversity(*args, **kwargs)` given a valid metric for `partial_beta_diversity` and an enumerated upper triangle of the distance matrix for `id_pairs`? In my opinion, that would be the most parsimonious API. However, I don't have strong feelings on this. 

(I realize that `validate=None` would make the above constraint not work)